test/e2e/upg_e2e.go

```diff
@@ -604,6 +604,69 @@ var _ = ginkgo.Describe("CLI debug commands", func() {
 
 // TODO: validate both binapi and CLI against each other
 var _ = ginkgo.Describe("UPG Binary API", func() {
+	ginkgo.Context("for upf ueip pool", func() {
+		f := framework.NewDefaultFramework(framework.UPGModeTDF, framework.UPGIPModeV4)
+
+		addPool := func(isAdd bool, nwi string, name string) {
+			nwiConverted := append(util.EncodeFQDN(nwi), 0)
+			nameConverted := append(util.EncodeFQDN(name), 0)
+			req := &upf.UpfUeipPoolNwiAdd{
+				IsAdd:      isAdd,
+				Identity:   nwiConverted,
+				NwiName:    nameConverted,
+				NwiNameLen: uint8(len(nameConverted) - 1),
+			}
+			reply := &upf.UpfUeipPoolNwiAddReply{}
+
+			gomega.Expect(
+				f.VPP.ApiChannel.SendRequest(req).ReceiveReply(reply),
+			).To(gomega.Succeed(), "upf_tdf_ul_enable_disable")
+		}
+
+		type poolPairing struct {
+			nwi  string
+			name string
+		}
+
+		dumpPools := func() []poolPairing {
+			reqCtx := f.VPP.ApiChannel.SendMultiRequest(&upf.UpfUeipPoolDump{})
+			var ret []poolPairing
+			for {
+				msg := &upf.UpfUeipPoolDetails{}
+				stop, err := reqCtx.ReceiveReply(msg)
+				gomega.Expect(err).To(gomega.BeNil())
+				if stop {
+					break
+				}
+
+				identity := strings.TrimRight(string(msg.Identity), "\x00")
+				nwiName := strings.TrimRight(string(msg.NwiName), "\x00")
+
+				ret = append(ret, poolPairing{
+					nwi:  util.DecodeFQDN([]byte(identity)),
+					name: util.DecodeFQDN([]byte(nwiName)),
+				})
```

Comment:
there is some confusion going on here:

* the identity is the PFCP "UE IP address Pool Identity", is a) not a `nwi` and b) a plain binary string without any special encoding
* nwiName is the PFCP "Network Instance", encoding rules are somewhat muddy, but it can be either a plain binary string or use DNS label encoding. `{En,De}codeFQDN` appears to only support the DNS label encoding (and not particularly good, but might be good enough for testing)